src/controller.cpp

```diff
@@ -104,38 +100,16 @@ bool Ctrl::runCalculation(){
       gui->generateRadiusMap(),
       gui->getIncludedElements(),
       gui->getMaxRad())){
-    data.success = false;
-    return data.success;
-  }
-
-  // process atom data for unit cell analysis if the option it ticked
-  if(gui->getAnalyzeUnitCell()){
-    if(!current_calculation->processUnitCell()){
-      data.success = false;
-      return data.success;
-    }
+    return false;
   }
-
-  // determine which atoms will be taken into account
-  current_calculation->setAtomListForCalculation();
-  // place atoms in a binary tree for faster access
-  current_calculation->storeAtomsInTree();
-  // set size of the box containing all atoms
-  current_calculation->defineCell();
-
-  current_calculation->generateChemicalFormula();
-
-  // assign voxel types and store the volume(s)
-  data = current_calculation->calcVolume();
-
-  // start the timer and save the total calculation time
+  auto start = std::chrono::steady_clock::now();
+  CalcReportBundle data = current_calculation->generateVolumeData();
   auto end = std::chrono::steady_clock::now();
-  double calc_time = std::chrono::duration<double>(end-start).count();
-  current_calculation->setTotalCalcTime(calc_time);

```

Comment:
This line sent the total calculation time to the model so it could be written in the report.
Without it, the report indicates 0 s.
Is there a reason for removing this line ?

Reply:
I find It strange that the controller should tell the model how long the calculation took, when the model is the one running the calculation. As far as I see there there is virtually no downside to using the calculation time already stored in _data inside the model class other than that a small bit of overhead is unaccounted for. As far as I see though, this isn't part of the time that is relevant when asking about the calculation time. I'll fix the issue with the output file.

Reply:
The goal is to tell how much time passed between the user pressing the button and getting the result.

Reply:
If that is the goal then the calc time shouldn't be stored inside the model class.

Reply:
Well, I still want to provide this time both in the notification box and the report file. The goal of the report bundle is to store all data useful for the report, isn't it ?

Reply:
Then how about this: Instead of counting everything towards the calculation time, we only count what happens inside the Model. This means what will not be included is the time it takes `setParameters()` to copy the inputs to the Model class and the time it takes for the output to be printed. IMO neither of these should be included in "calculation time" anyway and neither of them are significant anyway.

In turn I will add the time is takes the app to make the preparations for the calculation. I think it's useful to be able to time individual sections of the code separately, which is why I will store them in a std::vector. When the time is requested for the report or the output, the sum of all vector elements will be returned.

This way, all times are generated inside the Model class and everything (relevant) is accounted for. Your thoughts?

Reply:
Sounds good to me.

Reply:
I've implemented this now, I'd suggest you check once more that everything works. I saw that the report file now prints a time, my unit tests all works as intended, and the preparation time has also been added. Let me know if everything works, then I'll merge